Add missing recipes

johnjenkman opened the pull request:
Adding missing recipes from DIY Dog 2018.

johnjenkman committed:
Added missing recipes 240 to 250

Added Blitz Saison, Hello My Name Is Ingrid 2016, Crew Brew, Gin Blitz,
AB:21, Beatnik, Casino Rye Ale, Self Assembly Pope, Twin Atlantic,
Hop Shot and Rye IPA.

johnjenkman committed:
Fixed minor issue with invalid json

johnjenkman committed:
Added missing recipes 251 to 260

Added New England IPA, Paradox Rye, Semi Skimmed Occultist, Small Batch
90 Shilling, Small Batch Dortmunder, Small Batch Kellerbier, Small Batch
Small Batch Sorachi Ace Session, Small Batch Tripel and Tropic Thunder.

johnjenkman committed:
Added missing recipes 261 to 270

Added AB:22, Blonde Export Stout, Hazy Jane, Pump Action Poet, and the
following Small Batch Beers: East Coast Crush, Imperial Pale Weizen,
Lemon Meringue Pie, Mandarina Lager, Nitro Breakfast Stout.

johnjenkman committed:
Added missing recipes 271 to 280

Added AB:23, Make Earth Great Again, Nine To Five Wizard, Prototype
Black Rye IPA, Prototype Blonde Ale, Prototype Double IPA, Sidewalk
Shark, Slot Machine and the following Small Batch Beers - Cranachan
Cream Ale and Spelt & Honey Saison.

johnjenkman committed:
Added missing recipes 281 to 290

Added AB:24, Homicidal Puppet Helpdesk, Pina Colada Sidewalk and the
following Hello My Name Is recipes - Agnetha, Aune, Helga, Lieke,
Marianne, Niamh and Sari.

johnjenkman committed:
Added missing recipes 291 to 300

Added Choco Libre, Declassified Demo-God, East of Vermont, Hello My Name
 Is Maria, Hello My Name Is Sofia, I Wanna Be Your Dog, Indie Pale Ale,
Off-Duty Alien, Opaque Jake and Raspberry Popsicle Parade.

johnjenkman committed:
Updated README

johnjenkman committed:
Added recipes 301 to 305

Added Hazy Jane Bourbon and Rye barrel aged, Karma Cloud, Native Son and
Small Batch Dry-hopped Pilsner.

johnjenkman committed:
Added recipes 306 to 310

Added AB:25, Clockwork Tangering, Kamikaze Knitting Club, Paradox Grain,
 and Very Big Moose.

johnjenkman committed:
Added recipes 311 to 315

Added Baltic Fleet, Dog G, Mallow Mafia, Manic Mango and Sonic Boom.

johnjenkman committed:
Added recipes 316 to 320

Added Al Adjore! Grano Giusto, King of Eights, Neverland and SOS May Day

johnjenkman committed:
Added recipes 321 to 325

Added Fools Gold, Interstate Vienna, Jet Trash, Jinx and Zipcode.

johnjenkman commented:
@samjbmason all 325 beers have now been added with the changes in this pull request. Can you please review and merge?

Would it also be possible to create a 3.0 feature branch for some of the major refactoring such as correcting IDs to match DIY Dog, issue #43 etc.

sammdec commented on data/ab21.json:
```diff
@@ -0,0 +1,102 @@
+  "tagline": "Liquorice Blackcurrant Stout.",
+  "first_brewed": "2016",
+  "description": "AB:21 is a liquorice and blackcurrant infused imperial stout. Rolling in at an imposing 12% abv, this huge and decadent dark beer dials down the charred roastiness and focuses instead on the rich, full-bodied chocolate and dark fruit character of the style.",
+  "image_url": "",
```
There shouldn’t be any empty strings instead please use null where there is no value

johnjenkman replied:
All set and pushed

sammdec commented on data/ab22.json:
```diff
@@ -0,0 +1,163 @@
+        "unit": "celsius"
+      }
+    },
+    "twist": ""
```
Same rule applies here for empty strings. Please use null

johnjenkman replied:
All set and pushed

sammdec replied:
This value needs to be set as `null`

johnjenkman replied:
Apologies. All set now. Found a few others from older recipes and fixed them up too,

sammdec reviewed (changes requested):
This is brilliant work. Thank you so much for all the contribution. I did find one issue that I have left a comment on above.

johnjenkman committed:
Fixed invalid image_urls

Updated incorrect image_urls to contain a null string.

johnjenkman committed:
Fixed invalid image_urls

Updated incorrect image_urls to contain a null string. Minor typo.

johnjenkman reviewed (commented):
Updated multiple files to correct empty string and replaced with null string.

sammdec commented on data/ab21.json:
```diff
@@ -0,0 +1,102 @@
+  "tagline": "Liquorice Blackcurrant Stout.",
+  "first_brewed": "2016",
+  "description": "AB:21 is a liquorice and blackcurrant infused imperial stout. Rolling in at an imposing 12% abv, this huge and decadent dark beer dials down the charred roastiness and focuses instead on the rich, full-bodied chocolate and dark fruit character of the style.",
+  "image_url": "null",
```
This actually needs to be null rather than a string containing the characters "null"

johnjenkman replied:
Apologies. All set now. Found a few others from older recipes and fixed them up too,

sammdec replied:
HI there, this is still incorrect it needs to be like the suggestion below
```suggestion
  "image_url": null,
```

johnjenkman replied:
@samjbmason that causes the tests to fail. According to the schema it needs to be a string.

```console
[ { keyword: 'type',
    dataPath: '.image_url',
    schemaPath: '#/properties/image_url/type',
    params: { type: 'string' },
    message: 'should be string' } ]
```

I would suggest that schema change should be raised to address in v3 because there are other proposed changes that will mean also changing the schema.

johnjenkman replied:
@samjbmason All fixed in https://github.com/samjbmason/punkapi-db/pull/41/commits/c7bd8e9a7223d73f1c5f7c12166d6a1d40017567, https://github.com/samjbmason/punkapi-db/pull/41/commits/fb8f0d7da520b6b5ca2bca07ed0a2f4577e91a64 and https://github.com/samjbmason/punkapi-db/pull/41/commits/5e3a7c3e4721aabb69338e84f88e0888193229bc

sammdec commented on data/ab23.json:
```diff
@@ -0,0 +1,134 @@
+        "unit": "celsius"
+      }
+    },
+    "twist": ""
```
Should be `null`
```suggestion
    "twist": null
```

johnjenkman replied:
Apologies. All set now. Found a few others from older recipes and fixed them up too,

johnjenkman committed:
Fixed empty strings

Updated empty strings to contain a null string.

johnjenkman committed:
Fixed empty strings

Updated empty strings to contain a null string.

johnjenkman reviewed (commented):
Fixed up all remaining empty strings from this change set and also some empty strings in the current version.

sammdec reviewed (changes requested):
I made a suggestion below but it might need to be applied to more than one of the recipes

samjbmason commented:
Ah right I would say that was a bit presumptive in the schema.
On 25 Feb 2019, 22:49 +0000, johnjenkman <notifications@github.com>, wrote:
> @johnjenkman commented on this pull request.
> In data/ab21.json:
> > @@ -0,0 +1,102 @@
> +{
> +  "id": 244,
> +  "name": "AB:21",
> +  "tagline": "Liquorice Blackcurrant Stout.",
> +  "first_brewed": "2016",
> +  "description": "AB:21 is a liquorice and blackcurrant infused imperial stout. Rolling in at an imposing 12% abv, this huge and decadent dark beer dials down the charred roastiness and focuses instead on the rich, full-bodied chocolate and dark fruit character of the style.",
> +  "image_url": "null",
> @samjbmason that causes the tests to fail. It needs to be a string.
> [ { keyword: 'type', dataPath: '.image_url', schemaPath: '#/properties/image_url/type', params: { type: 'string' }, message: 'should be string' } ]
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

johnjenkman commented:
@samjbmason that schema issue has been there a long tine. Do you want me to fix the schema and update the null values as part of this pull request. Or should the schema issue be corrected in V3 as part of a full schema review as we know the schema will need to be updated in V3 anyway as part of implementing #43

samjbmason commented:
Yeah I imagine the issue has probs been there since the beginning I think it was because I assumed we should always have images for beers but I think the best thing to do would be to update the schema to allow null values, I think we can do it in version 2 and we will then do V3 with the big schema change

johnjenkman committed:
Added null support to imague_url

Updated schema.json to allow support for null value in image_url.

johnjenkman committed:
Updated image_url null value

Amended image_url from null string to a null value after adding support
for null values in the schema.

johnjenkman committed:
Updated twist null value

Corrected twist to fix change null string to a null value.

johnjenkman reviewed (commented):
@samjbmason all remaining null issues should now be fixed in https://github.com/samjbmason/punkapi-db/pull/41/commits/5e3a7c3e4721aabb69338e84f88e0888193229bc, https://github.com/samjbmason/punkapi-db/pull/41/commits/fb8f0d7da520b6b5ca2bca07ed0a2f4577e91a64 and https://github.com/samjbmason/punkapi-db/pull/41/commits/c7bd8e9a7223d73f1c5f7c12166d6a1d40017567 I thought I made a comment the other day but I can't seem to find it.
